linetools/isgm/utils.py

```diff
@@ -451,4 +451,81 @@ def coincident_components(comp1, comp2, tol=0.2*u.arcsec):
             overlap = line1.coincident_line(line2)
             if overlap is True:
                 return True
-    return False
\ No newline at end of file
+    return False
+
+
+def group_coincident_compoments(comp_list):
+    """For a given input list of component, this function
+    group together components that are coincident to each other
+    (including by transitivity), and return them as a list of
+    component lists.
+
+    Parameters
+    ----------
+    comp_list : list of AbsComponent
+        Input list of components to group
+
+    Returns
+    -------
+    output : list of lists of AbsComponent
+        The grouped components as individual lists
+        in the output list.
+    """
+    # the first extreme case is that all components are independent
+    # of each other, in which case we have the following output shape
+    out = [[] for kk in range(len(comp_list))]
+
+    for ii in range(len(comp_list)):
```

Comment:
Do not how to do the enumerate for the `jj` index in current line 491, as `jj` has to start from `ii+1`. Will leave it as is for clarity.